backend/src/routes/search.rs

```diff
@@ -0,0 +1,102 @@
+use rocket::{Build, Rocket};
+use rocket_db_pools::Connection;
+
+use crate::pool::{Search, TypesenseSearch};
+use crate::req::pulsar::*;
+
+use serde_json::json;
+
+pub async fn init(rocket: Rocket<Build>) -> Rocket<Build> {
+    rocket.mount(
+        "/search",
+        routes![
+            search_burrow_keyword,
+            search_burrow_id,
+            search_post_keyword,
+            search_post_id
+        ],
+    )
+}
+///Search keyword in burrow title/introduction.
+#[get("/burrow/<keyword>")]
+async fn search_burrow_keyword(searchpool: Connection<TypesenseSearch>, keyword: &str) -> String {
+    let client = searchpool.into_inner();
+    let uri = format!("/collections/burrows/documents/search?q={}&query_by=title,introduction&filter_by=&sort_by=update_time", keyword);
+    let response = match client.build_get(&uri).send().await {
+        Ok(a) => a.json::<serde_json::Value>().await.unwrap().to_string(),
+        Err(e) => return format!("build_get send Error: {:?}", e),
+    };
+    let result: SearchResult = serde_json::from_str(&response).unwrap();
+    match result.found {
+        0 => "No results!".to_string(),
+        _ => json!({
+            "found":result.found,
+            "hits":result.hits
+        })
+        .to_string(),
+    }
+}
+///Search burrow with id.
+#[get("/burrow/<burrow_id>")]
+async fn search_burrow_id(searchpool: Connection<TypesenseSearch>, burrow_id: i64) -> String {
+    let client = searchpool.into_inner();
+    let uri = format!("/collections/burrows/documents/search?q={}&query_by=burrow_id&filter_by=&sort_by=update_time", burrow_id);
+    let response = match client.build_get(&uri).send().await {
+        Ok(a) => a.text().await.unwrap(),
+        Err(e) => return format!("build_get send Error: {:?}", e),
+    };
+    let result: SearchResult = serde_json::from_str(&response).unwrap();
+    match result.found {
+        0 => "No results!".to_string(),
+        _ => json!({
+            "found":result.found,
+            "hits":result.hits
+        })
+        .to_string(),
+    }
```

Comment:
when given only burrowid, use sea-orm to search postgres, no need to touch typesense